ddtrace/llmobs/_integrations/langchain.py

```diff
@@ -260,6 +263,52 @@ def _llmobs_set_meta_tags_from_embedding(
             except (TypeError, IndexError):
                 log.warning("Failed to write output vectors", output_embedding)
 
+    def _llmobs_set_meta_tags_from_similarity_search(
+        self,
+        span: Span,
+        input_query: str,
+        output_documents: Union[List[Any], None],
+        error: bool = False,
+        is_workflow: bool = False,
+    ) -> None:
+        span.set_tag_str(SPAN_KIND, "workflow" if is_workflow else "retrieval")
+        span.set_tag_str(MODEL_NAME, span.get_tag(MODEL) or "")
+        span.set_tag_str(MODEL_PROVIDER, span.get_tag(PROVIDER) or "")
+
+        input_tag_key = INPUT_VALUE
+        output_tag_key = OUTPUT_VALUE if is_workflow else OUTPUT_DOCUMENTS
+
+        if input_query is not None:
+            try:
+                formatted_inputs = self.format_io(input_query)
+                if isinstance(formatted_inputs, str):
+                    span.set_tag_str(input_tag_key, formatted_inputs)
+                else:
+                    span.set_tag_str(input_tag_key, json.dumps(formatted_inputs))
+            except TypeError:
+                log.warning("Failed to serialize similarity search input to JSON")
+        if error:
+            span.set_tag_str(output_tag_key, "")
+        elif isinstance(output_documents, list):
+            if is_workflow:
+                span.set_tag_str(output_tag_key, json.dumps(output_documents))
+            else:
+                try:
+                    documents = [
+                        Document(
+                            id=d.id,
```

Comment:
It seems like ID is optional so let's default to an empty string if document.id is None.